Foundation/src/DirectoryWatcher.cpp

```diff
@@ -376,6 +375,7 @@ class LinuxDirectoryWatcherStrategy: public DirectoryWatcherStrategy
 
 private:
 	int _fd;
+	fd_set fds;
```

Comment:
Why move here?

Reply:
> but this is not https://clickhouse-test-reports.s3.yandex.net/25969/ddf9e524cc1012271f433cb526bc921acd7c5376/stress_test_(thread)/stderr.log
> 
> but they look related to the same problem

Yes, both it and https://clickhouse-test-reports.s3.yandex.net/25969/127a0f35910a097b5240d68cc60f5e30b9e697ac/stress_test_(address)/stderr.log ref to the loc of `FD_SET(_fd, &fds)`, but looks like it also a false positive, so I move `fds` to class member.

Reply:
But still need to do something with https://clickhouse-test-reports.s3.yandex.net/25969/ddf9e524cc1012271f433cb526bc921acd7c5376/stress_test_(thread)/stderr.log

Reply:
Looks like maybe it also because the thread is still running after the the Strategy object destructed. I move it to behind strategy, let's see does it work.